src/lib/components/error-boundary.svelte

```diff
@@ -0,0 +1,44 @@
+<script lang="ts">
+  import { networkError } from '$lib/stores/error';
+  import { isNetworkError } from '$lib/utilities/is-network-error';
+  import Error from './error.svelte';
+
+  export let error = null;
+  export let onError = null;
+
+  let theError;
+
+  let status = 500;
+
+  $: {
+    if (error && $error) {
+      theError = error;
+    }
+    if (networkError && $networkError) {
+      theError = networkError;
+    }
+
+    if (onError && theError) {
+      onError(theError);
+    }
+
+    if (isNetworkError($theError)) {
+      status = $theError.statusCode;
+    }
+  }
```

Comment:
@stevekinney This is where we check the error that is thrown into the error boundary or check the error store to react to an error event during runtime.